TypeCobol.Test/Parser/Programs/Cobol85/RedefinesWithOccurs.Mix.txt

```diff
@@ -0,0 +1,535 @@
+      *TODO Wait for support of UTF-8 (#2504)
+      *01 Group-C02 group-usage is UTF-8.
+      *    05 Array-C02-KO.
+      *    05 Array-C02-KO-R redefines Array-C02-KO OCCURS 1.
+      
+      
+      *********************************************
+      *D. Usage is not declared directly on parent (test recursivity)
+      *********************************************
+      
+       01 Group-D1 usage is pointer.
+           05 Group-D11.
+                10 Array-D1-OK.
+                10 Array-D1-OK-R redefines Array-D1-OK OCCURS 1.
+      
+      
+      *The"PICTURE"clause for item"ARRAY-D2-KO"was not compatible with
+      *the specified"USAGE"."USAGE DISPLAY"was assumed.
+      *A"PICTURE"clause was not found for elementary item"ARRAY-D2-KO".
+      *   "PICTURE X(1)"was assumed.
+       01 Group-D2 comp-5.
+           05 Group-D21.
+Line 416[20,30] <27, Error, Syntax> - Syntax error : A group item cannot be empty. Add children, picture or usage declaration.
+                10 Array-D2-KO.
+      *The "PICTURE" clause for item "ARRAY-D2-KO-R" was not compatible
+      * with the specified "USAGE".  "USAGE DISPLAY" was assumed.
+      *A "PICTURE" clause was not found for elementary item
+      *"ARRAY-D2-KO-R". "PICTURE X(1)" was assumed.
+Line 421[20,32] <27, Error, Syntax> - Syntax error : A group item cannot be empty. Add children, picture or usage declaration.
+                10 Array-D2-KO-R redefines Array-D2-KO OCCURS 1.
+      
+      
+     
+      
+       01 Group-D3 usage is procedure-pointer.
+           05 Group-D31.
+                10 Group-D311.
+                    15 Array-D3-OK.
+                    15 Array-D3-OK-R redefines Array-D3-OK OCCURS 1.
+      
+      
+       01 Group-D4 usage is display-1.
+           05 Group-D41.
+                10 Group-D411.
+      
+      *The"PICTURE"clause for item"ARRAY-D4-KO"was not compatible with
+      *the specified"USAGE"."USAGE DISPLAY"was assumed.
+      *A"PICTURE"clause was not found for elementary item"ARRAY-D4-KO".
+      *   "PICTURE X(1)"was assumed.
+Line 441[24,34] <27, Error, Syntax> - Syntax error : A group item cannot be empty. Add children, picture or usage declaration.
+                    15 Array-D4-KO.
+      
+      *The "PICTURE" clause for item "ARRAY-D4-KO-R" was not compatible
+      * with the specified "USAGE".  "USAGE DISPLAY" was assumed.
+      *A "PICTURE" clause was not found for elementary item
+      *"ARRAY-D4-KO-R". "PICTURE X(1)" was assumed.
+Line 447[24,36] <27, Error, Syntax> - Syntax error : A group item cannot be empty. Add children, picture or usage declaration.
+                    15 Array-D4-KO-R redefines Array-D4-KO OCCURS 1.
+      
+      
+      
+      
+      
+       01 Group-D5 usage is comp-1.
+           05 Array-D5-OK.
+      *The specified "USAGE" was different from the "USAGE" specified at
+      *the group level.  The group "USAGE" was assumed for this item.
+           05 Group-D51 group-usage is national.
```

Comment:
So not supported yet ? Could you explain a bit more the expected behavior of our parser in comment as we get different errors than with IBM compiler ?

Reply:
Comment changed. Is it better.